### ISSUE TITLE
The release tar does not include the `hdmf-common-schema` sub module

### ISSUE THREAD

**Issue by sanjayankur31** [hdmf-dev/hdmf#195]
The release tarball does not include any files in the `hdmf-common-schema` directory:

```
$ tar -tvf hdmf-1.3.3.tar.gz | grep hdmf-common-schema
drwxrwxr-x root/root         0 2019-10-16 00:47 hdmf-1.3.3/src/hdmf/common/hdmf-common-schema/

```

I *think* the files there are required to run the test-suite, so they should be included too maybe?

```
        |  Test name  | Output |    Status   |
        +-------------+--------+-------------+
        | test_sparse |        | [0;0;31merror       |
        | test_table  |        | [0;0;31merror       |
        +-------------+--------+-------------+
  [0;0;31mERROR in test common.test_sparse (unittest2.loader._FailedTest):
  Traceback (most recent call last):
  ImportError: Failed to import test module: common.test_sparse
  Traceback (most recent call last):
    File "/usr/lib/python3.8/site-packages/unittest2/loader.py", line 456, in _find_test_path
      module = self._get_module_from_name(name)
    File "/usr/lib/python3.8/site-packages/unittest2/loader.py", line 395, in _get_module_from_name
      __import__(name)
    File "/builddir/build/BUILD/hdmf-1.3.3/tests/unit/common/test_sparse.py", line 1, in <module>
      from hdmf.common import CSRMatrix
    File "/builddir/build/BUILDROOT/python-hdmf-1.3.3-4.fc32.x86_64/usr/lib/python3.8/site-packages/hdmf/common/__init__.py", line 108, in <module>
      raise RuntimeError("Unable to load a TypeMap - no namespace file found")
  RuntimeError: Unable to load a TypeMap - no namespace file found
  [0;0;31mERROR in test common.test_table (unittest2.loader._FailedTest):
  Traceback (most recent call last):
  ImportError: Failed to import test module: common.test_table
  Traceback (most recent call last):
    File "/usr/lib/python3.8/site-packages/unittest2/loader.py", line 456, in _find_test_path
      module = self._get_module_from_name(name)
    File "/usr/lib/python3.8/site-packages/unittest2/loader.py", line 395, in _get_module_from_name
      __import__(name)
    File "/builddir/build/BUILD/hdmf-1.3.3/tests/unit/common/test_table.py", line 3, in <module>
      from hdmf.common import DynamicTable, VectorData, ElementIdentifiers, DynamicTableRegion
    File "/builddir/build/BUILDROOT/python-hdmf-1.3.3-4.fc32.x86_64/usr/lib/python3.8/site-packages/hdmf/common/__init__.py", line 108, in <module>
      raise RuntimeError("Unable to load a TypeMap - no namespace file found")
  RuntimeError: Unable to load a TypeMap - no namespace file found
```

**Comment by sanjayankur31** [comment]
Yes, manually downloading the tar and copying the files in seems to get the tests to run correctly.

**Comment by rly** [comment]
Hi @sanjayankur31 , thank you for the bug report and I am glad you were able to find a workaround.

The source code tarball from the GitHub releases page and the source code ZIP from the main repo page > "Clone or Download" button > Download ZIP button are both missing the `hdmf-common-schema` submodule. These archives are automatically generated by GitHub using the `git archive` command, which unfortunately has no option to include submodules. 

Including submodules as part of the source code tarball/zip is a feature request on GitHub: https://github.community/t5/How-to-use-Git-and-GitHub/Create-Release-that-contains-submodule/td-p/12073 However, it looks like it will probably not be implemented anytime soon. I have added a +1 comment there.

I will look into a way to add a downloadable single file asset to future releases that includes the source code and submodule contents. However, the last two assets ("Source code .zip", "Source code .tar.gz") on the GitHub release page are automatically generated by GitHub and will always be missing the submodule contents.

For now, if you want to download the source code with submodules, I recommend using `git clone  --recurse-submodules git@github.com:hdmf-dev/hdmf.git` for the latest or `git clone  --recurse-submodules --branch 1.3.3 git@github.com:hdmf-dev/hdmf.git` for version 1.3.3.

**Comment by sanjayankur31** [comment]
Thanks for your reply @rly . I figured it was a github thing---it doesn't include sub-modules in the release tar (I'm not sure if git does either if one uses git-archive).

When building for Fedora, we can't use `git clone`---the sources must be downloaded by us maintainers and then checked-in. I've worked around it by including the sub-module as another source in the spec file. That will work for us:

https://src.fedoraproject.org/rpms/python-hdmf/blob/master/f/python-hdmf.spec#_27

**Comment by Aniket-Pradhan** [comment]
Hello hdmf dev team and @sanjayankur31 

I was updating the python-hdmf package for Fedora, and the [source](https://github.com/hdmf-dev/hdmf/releases/download/1.4.0/hdmf-1.4.0.tar.gz) provided on the releases page can now be used, as it contains the required schema files. Thanks a lot for creating a separate tar. :smile: 
 
There is, however, one small issue. The release tar does not contain the [Legal.txt](https://github.com/hdmf-dev/hdmf/blob/dev/Legal.txt) file. I think that Legal.txt should be provided with the release tar. As for now, I have included it as a separate source. I guess that file can be covered in the next release tar.

**Comment by rly** [comment]
> There is, however, one small issue. The release tar does not contain the [Legal.txt](https://github.com/hdmf-dev/hdmf/blob/dev/Legal.txt) file. I think that Legal.txt should be provided with the release tar. As for now, I have included it as a separate source. I guess that file can be covered in the next release tar.

Legal.txt is now included in the release tarball in 1.5.0 and moving forward. Thanks!
